Tests/WitsmlExplorer.Api.Tests/Workers/AnalyzeGapWorkerTests.cs

```diff
@@ -0,0 +1,271 @@
+            .Returns((WitsmlLogs logs, OptionsIn options) => isLogDataWithGaps
+                ? Task.FromResult(GetTestWitsmlLogs(GetTestLogDataWithGaps(isDepthLog),isDepthLog))
+                : Task.FromResult(GetTestWitsmlLogs(GetTestLogDataWithoutGaps(isDepthLog), isDepthLog)));
+
+        witsmlClient.InSequence(mockSequence)
+            .Setup(client =>
+                client.GetFromStoreAsync(It.IsAny<WitsmlLogs>(), It.IsAny<OptionsIn>()))
+            .Returns(Task.FromResult(new WitsmlLogs(){Logs = new List<WitsmlLog>() }));
+    }
+    
+    private static WitsmlLogs GetTestWitsmlLogs(WitsmlLogData logData, bool isDepthLog)
+    {
+        return new WitsmlLogs
+        {
+            Logs = new WitsmlLog
+            {
+                UidWell = WellUid,
+                UidWellbore = WellboreUid,
+                Uid = LogUid,
+                StartIndex = new WitsmlIndex(new DepthIndex(51, "m")),
+                EndIndex = new WitsmlIndex(new DepthIndex(54, "m")),
+                StartDateTimeIndex = "2023-04-19T00:00:00Z",
+                EndDateTimeIndex = "2023-04-19T00:00:02Z",
+                IndexCurve = new WitsmlIndexCurve() { Value = "Depth" },
+                LogData = logData,
+                Direction = WitsmlLog.WITSML_DIRECTION_INCREASING,
+                IndexType = isDepthLog ? WitsmlLog.WITSML_INDEX_TYPE_MD : WitsmlLog.WITSML_INDEX_TYPE_DATE_TIME,
+                LogCurveInfo = new List<WitsmlLogCurveInfo>()
+                {
+                    new WitsmlLogCurveInfo()
+                    {
+                        Mnemonic = "Depth",
+                        MinIndex = new WitsmlIndex(new DepthIndex(51, "m")),
+                        MaxIndex = new WitsmlIndex(new DepthIndex(54, "m")),
+                        MinDateTimeIndex = "2023-04-19T00:00:00Z",
+                        MaxDateTimeIndex = "2023-04-19T00:00:03Z",
+                        
+                    },
+                    new WitsmlLogCurveInfo()
+                    {
+                        Mnemonic = "BPOS",
+                        MinIndex = new WitsmlIndex(new DepthIndex(51, "m")),
+                        MaxIndex = new WitsmlIndex(new DepthIndex(54, "m")),
+                        MinDateTimeIndex = "2023-04-19T00:00:00Z",
+                        MaxDateTimeIndex = "2023-04-19T00:00:03Z"
+                    },
+                    new WitsmlLogCurveInfo()
+                    {
+                        Mnemonic = "SPM1",
+                        MinIndex = new WitsmlIndex(new DepthIndex(51, "m")),
+                        MaxIndex = new WitsmlIndex(new DepthIndex(54, "m")),
+                        MinDateTimeIndex = "2023-04-19T00:00:00Z",
+                        MaxDateTimeIndex = "2023-04-19T00:00:03Z"
+                    },
+                },
+            }.AsSingletonList()
+        };
+    }
+    
+    private static WitsmlLogData GetTestLogDataWithoutGaps(bool isDepthLog)
```

Comment:
Not a must, but would be nice to have a test or two for decreasing logs, especially as it's not working for me now. Simplest way to get the data would be to just reverse the Data list, and set Decreasing as direction.

Reply:
Added unit tests for decreasing direction.

Reply:
Super!